src/Microsoft.Data.SqlClient/tests/ManualTests/DataCommon/DataTestUtility.cs

```diff
@@ -1073,6 +1073,26 @@ public static string GetMachineFQDN(string hostname)
             return fqdn.ToString();
         }
 
+        public static bool IsNotLocalhost()
+        {
+            // get the tcp connection string
+            SqlConnectionStringBuilder builder = new(DataTestUtility.TCPConnectionString);
+
+            string hostname = "";
+
+            // parse the datasource
+            ParseDataSource(builder.DataSource, out hostname, out _, out _);
+
+            // hostname must not be localhost, ., 127.0.0.1 nor ::1
+            return !(new string[] { "localhost", ".", "127.0.0.1", "::1" }).Contains(hostname.ToLowerInvariant());
+
+        }
+
+        public static bool IsKerberosManagedSNI() 
+        {
+            return (AreConnStringsSetup() && IsUsingManagedSNI() && IsNotLocalhost() && IsKerberosTest && IsNotAzureServer() && IsNotAzureSynapse());
+        }
+
```

Comment:
That's not quite what I meant. I meant for this to be in the test class itself since it's very specific to the test's requirements.